rootston/output.c

```diff
@@ -512,6 +518,25 @@ static void render_output(struct roots_output *output) {
 
 damage_finish:
 	pixman_region32_fini(&damage);
+
+	// Send frame done events to all surfaces
+	if (output->fullscreen_view != NULL) {
+		view_for_each_surface(view, surface_send_frame_done, &now);
+
+#ifdef WLR_HAS_XWAYLAND
+		if (view->type == ROOTS_XWAYLAND_VIEW) {
+			xwayland_children_for_each_surface(view->xwayland_surface,
+				surface_send_frame_done, &now);
+		}
+#endif
+	} else {
+		struct roots_view *view;
+		wl_list_for_each_reverse(view, &desktop->views, link) {
+			view_for_each_surface(view, surface_send_frame_done, &now);
```

Comment:
This should check that the surface actually intersects the output.

Reply:
Ah, you're right - for some reason I was believing `view_for_each_surface` filtered them.